nym-api/src/support/storage/manager.rs

```diff
@@ -51,6 +52,25 @@ impl AvgGatewayReliability {
     }
 }
 
+// Helper struct for in-memory state during calculation for an interval
+#[derive(Debug, Default, Clone)]
+struct NodeCorrectionIntervalState {
+    pos_samples: u32,
+    neg_samples: u32,
+    fail_seq: u32,
+}
+
+// Output struct for the calculated corrected reliability for an interval
+#[derive(Debug, serde::Serialize)] // Add utoipa::ToSchema if this struct is directly exposed via API
+pub struct CorrectedNodeIntervalReliability {
```

Comment:
should the correction be done by nym  api or rather NM? (ideally, in stage 2, NM would be submitting corrected results to the contract)